utils/exportV1/src/main/java/io/apicurio/registry/utils/export/Export.java

```diff
@@ -0,0 +1,217 @@
+import java.util.zip.ZipOutputStream;
+
+import javax.inject.Inject;
+
+import org.apache.commons.codec.digest.DigestUtils;
+
+import io.apicurio.registry.client.RegistryRestClient;
+import io.apicurio.registry.client.RegistryRestClientFactory;
+import io.apicurio.registry.content.ContentHandle;
+import io.apicurio.registry.content.canon.ContentCanonicalizer;
+import io.apicurio.registry.rest.beans.Rule;
+import io.apicurio.registry.rest.beans.VersionMetaData;
+import io.apicurio.registry.types.ArtifactType;
+import io.apicurio.registry.types.RuleType;
+import io.apicurio.registry.types.provider.ArtifactTypeUtilProvider;
+import io.apicurio.registry.types.provider.ArtifactTypeUtilProviderFactory;
+import io.apicurio.registry.utils.IoUtil;
+import io.quarkus.runtime.QuarkusApplication;
+import io.quarkus.runtime.annotations.QuarkusMain;
+
+/**
+ * @author Fabian Martinez
+ */
+@QuarkusMain
+public class Export implements QuarkusApplication {
+
+    static {
+        //Workaround, because this app depends on apicurio-registry-app we are spawning an http server. Here we are setting the http port to a less probable used port
+        System.setProperty("quarkus.http.port", "9573");
+    }
+
+    @Inject
+    ArtifactTypeUtilProviderFactory factory;
+
+    /**
+     * @see io.quarkus.runtime.QuarkusApplication#run(java.lang.String[])
+     */
+    @Override
+    public int run(String... args) throws Exception {
+
+        if (args.length == 0) {
+            System.out.println("Missing required argument, registry url");
+            return 1;
+        }
+
+        String url = args[0];
+
+        Map<String, Object> conf = new HashMap<>();
+
+        if (args.length > 2 && args[1].equals("--client-props")) {
+            String[] clientconf = Arrays.copyOfRange(args, 2, args.length);
+            conf = Arrays.asList(clientconf)
+                .stream()
+                .map(keyvalue -> keyvalue.split("="))
+                .collect(Collectors.toMap(kv -> kv[0], kv -> kv[1]));
+            System.out.println("Parsed client properties " + conf);
+        }
+
+        RegistryRestClient client = RegistryRestClientFactory.create(url, conf);
+
+        File output = new File("registry-export.zip");
+        try (FileOutputStream fos = new FileOutputStream(output)) {
+
+            System.out.println("Exporting registry data to " + output.getName());
+
+            ZipOutputStream zip = new ZipOutputStream(fos, StandardCharsets.UTF_8);
+            EntityWriter writer = new EntityWriter(zip);
+            AtomicInteger errorCounter = new AtomicInteger(0);
+
+            Function<Entity, Void> exporter = entity -> {
+                try {
+                    writer.writeEntity(entity);
+                } catch (Exception e) {
+                    // TODO do something interesting with this
+                    e.printStackTrace();
+                    errorCounter.incrementAndGet();
+                }
+                return null;
+            };
```

Comment:
Probably not needed - just call the writer directly.

Reply:
wrapping the error sounds useful

Reply:
Don't we just want to throw the error out of the CLI tool if we get one?  The error is wrapped and counted on the server because we can't just quit and report an error when streaming an HTTP response.  But we **can** do that in a CLI.